core/domain/user_query_services.py

```diff
@@ -74,20 +195,19 @@ def send_email_to_qualified_users(
         email_intent: str. Intent of the email.
         max_recipients: int. Maximum number of recipients send emails to.
     """
-    query_model = user_models.UserQueryModel.get(query_id)
-    query_model.query_status = feconf.USER_QUERY_STATUS_ARCHIVED
-    recipient_ids = query_model.user_ids
+    user_query = get_user_query(query_id, strict=True)
+    recipient_ids = user_query.user_ids
 
     if max_recipients:
         recipient_ids = recipient_ids[:max_recipients]
 
     bulk_email_model_id = email_manager.send_user_query_email(
-        query_model.submitter_id, recipient_ids, email_subject,
-        email_body, email_intent)
-    query_model.sent_email_model_id = bulk_email_model_id
-    query_model.deleted = True
-    query_model.update_timestamps()
-    query_model.put()
+        user_query.submitter_id, recipient_ids, email_subject,
+        email_body, email_intent
+    )
+
+    user_query.delete(sent_email_model_id=bulk_email_model_id)
+    _save_user_query(user_query)
```

Comment:
The semantics here are weird. You wouldn't save something after deleting it. When you "delete" it, it's already gone.

Do you want something like mark_as_deleted() instead? Or obsolesce()/retire()?